proxy/codecs.go

```diff
@@ -78,9 +78,9 @@ func (m *partialExecute) GetOpCode() primitive.OpCode {
 	return primitive.OpCodeExecute
 }
 
-func (m *partialExecute) Clone() message.Message {
+func (m *partialExecute) DeepCopyMessage() message.Message {
 	return &partialExecute{
-		queryId: primitive.CloneByteSlice(m.queryId),
+		queryId: append(m.queryId[:0:0], m.queryId...),
```

Comment:
I'm not sure if this is the most efficient way of copying a byte slice... I'd think something like this would be better:

```
newByteSlice := make([]byte, len(m.queryId))
copy(newByteSlice, m.queryId)
return &partialExecute{queryId: newByteSlice)
```

This is also done on the `partialBatch` type below. And it is also what is done by the generated deepcopy function on the library: https://github.com/datastax/go-cassandra-native-protocol/blob/main/message/deepcopy_generated.go#L329-L333

Reply:
Maybe the end result is the same but I think this alternative version reads better.